xcdat/regridder/grid.py

```diff
@@ -503,3 +511,32 @@ def create_grid(
     grid = grid.bounds.add_missing_bounds()
 
     return grid
+
+
+def _validate_grid_has_single_axis_dim(
+    axis: CFAxisKey, coord_var: Union[xr.DataArray, xr.Dataset]
+):
+    """Validates that the grid's axis has a single dimension.
+
+    If the grid has multiple dimensions (e.g., "lat" and "latitude" dims), xcdat
+    cannot interpret which one to use for grid operations. If ``coord_var`` is
+    an ``xr.Dataset``, the grid has multiple dimensions.
+
+    Parameters
+    ----------
+    axis : CFAxisKey
+        The CF axis key ("X", "Y", "T", or "Z").
+    coord_var : Union[xr.DataArray, xr.Dataset]
+        The dimension coordinate variable(s) for the axis.
+
+    Raises
+    ------
+    ValueError
+        If the grid has multiple dimensions.
+    """
+    if isinstance(coord_var, xr.Dataset):
+        raise ValueError(
+            f"Multiple '{axis}' axis dims were found in this dataset, "
+            f"{list(coord_var.dims)}. Please drop the unused dimension(s) before "
+            "performing grid operations."
+        )
```

Comment:
As mentioned above, grids created in xcdat have a single dimension per axis.

`_validate_grid_has_single_axis_dim()` is specifically used for validation and type checking.